src/docgen/transpile/java.ts

```diff
@@ -0,0 +1,257 @@
+    };
+  }
+
+  public struct(struct: reflect.InterfaceType): transpile.TranspiledStruct {
+    const type = this.type(struct);
+    const inputs = struct.allProperties.map((p) =>
+      this.formatBuilderMethod(this.property(p)),
+    ).filter(isNotNull);
+    return {
+      type: type,
+      name: struct.name,
+      import: this.formatImport(type),
+      initialization: this.formatStructInitialization(type, inputs),
+    };
+  }
+
+  public interface(
+    iface: reflect.InterfaceType,
+  ): transpile.TranspiledInterface {
+    return {
+      name: iface.name,
+      type: this.type(iface),
+    };
+  }
+
+  public parameter(
+    parameter: reflect.Parameter,
+  ): transpile.TranspiledParameter {
+    return {
+      name: parameter.name,
+      parentType: this.type(parameter.parentType),
+      typeReference: this.typeReference(parameter.type),
+      optional: parameter.optional,
+    };
+  }
+
+  public property(property: reflect.Property): transpile.TranspiledProperty {
+    return {
+      name: property.name,
+      parentType: this.type(property.parentType),
+      typeReference: this.typeReference(property.type),
+      optional: property.optional,
+    };
+  }
+
+  public enum(enu: reflect.EnumType): transpile.TranspiledEnum {
+    return {
+      fqn: this.type(enu).fqn,
+      name: enu.name,
+    };
+  }
+
+  public enumMember(em: reflect.EnumMember): transpile.TranspiledEnumMember {
+    return {
+      fqn: `${this.enum(em.enumType).fqn}.${em.name}`,
+      name: em.name,
+    };
+  }
+
+  public unionOf(_types: string[]): string {
+    return 'java.lang.Object';
+  }
+
+  public listOf(type: string): string {
+    return `java.util.List<${type}>`;
+  }
+
+  public mapOf(type: string): string {
+    return `java.util.Map<java.lang.String, ${type}>`;
+  }
+
+  public any(): string {
+    return 'java.lang.Object';
+  }
+
+  public str(): string {
+    return 'java.lang.String';
+  }
+
+  public number(): string {
+    return 'java.lang.Number';
+  }
+
+  public boolean(): string {
+    return 'java.lang.Boolean';
+  }
+
+  public json(): string {
+    return 'com.fasterxml.jackson.databind.node.ObjectNode';
+  }
+
+  public date(): string {
+    return 'java.time.Instant';
+  }
+
+  public readme(readme: string): string {
+    return readme;
+  }
+
+  private formatImport(type: transpile.TranspiledType): string {
+    return `import ${type.fqn};`;
+  };
+
+  private formatParameters(
+    transpiled: transpile.TranspiledParameter | transpile.TranspiledProperty,
+  ): string {
+    const tf = transpiled.typeReference.toString({
+      typeFormatter: (t) => t.name,
+    });
+    return `${transpiled.name}: ${tf}`;
+  }
+
+  private formatArguments(inputs: string[]): string {
+    return inputs.join(', ');
+  };
+
+  private formatStructInitialization(type: transpile.TranspiledType, inputs: string[]): string {
+    const builder = `${type.name} ${toCamelCase(type.name)} = new ${type.name}.Builder()`;
```

Comment:
This ends up being a little bit tricky since the builder classes aren't part of the jsii type system, so it would be a little bit like re-implementing parts of jsii-srcmak I think.

I don't think this is necessary for the initial Java API reference experience - most Java users are familiar with fluent interfaces, and we're now showing code examples of how to use the builder classes.